logstash-core/lib/logstash/webserver.rb

```diff
@@ -8,19 +8,20 @@
 module LogStash 
   class WebServer
 
+    class HostBindingError < StandardError; end
+
     extend Forwardable
 
     attr_reader :logger, :status, :config, :options, :cli_options, :runner, :binder, :events
 
     def_delegator :@runner, :stats
 
-    DEFAULT_HOST = "127.0.0.1".freeze
```

Comment:
ah! now I understand, you prefer to actually make puma up and down, might try that, however seems a bit too much (bring puma up and down) for the whole thing, will see If I see a simpler way to implement this.